File: Iniciante/PHP/exercise_1049.php
<?php

function getAnimal ($filo, $classe, $alimentacao) {
    $animal = "nenhum encontrado";
    
    switch ($filo) {
        case 'vertebrado':
            switch ($classe) {
                case "ave":
                    switch ($alimentacao) {
                        case "carnivoro":
                            $animal = "aguia";
                            break;
                        case "onivoro":
                            $animal = "pomba";
                            break;
                    }
                    break;
                case "mamifero":
                    switch ($alimentacao) {
                        case "onivoro":
                            $animal = "homem";
                            break;
                        case "herbivoro":
                            $animal = "vaca";
                            break;
                    }
                    break;
            }
            break;
        case 'invertebrado':
            switch ($classe) {
                case "inseto":
                    switch ($alimentacao) {
                        case "hematofago":
                            $animal = "pulga";
                            break;
                        case "herbivoro":
                            $animal = "lagarta";
                            break;
                    }
                    break;
                case "anelideo":
                    switch ($alimentacao) {
                        case "hematofago":
                            $animal = "sanguessuga";
                            break;
                        case "onivoro":
                            $animal = "minhoca";
                            break;
                    }
                    break;
            }
            break;
        default:
            $animal = "não localizado";
    }

    return $animal;
}

$filo = readline("");
$classe = readline("");
$alimentacao = readline("");

$animal = getAnimal($filo, $classe, $alimentacao);

print_r($animal."\n");

//Concluído

File: Iniciante/Python/exercises from 1000 to 1099/exercise_1049.py
vertebra = str(input())
tipo = str(input())
alimentacao = str(input())

resultados = {
    'vertebrado': {
        'ave': {
            'carnivoro': 'aguia',
            'onivoro': 'pomba'
        },
        'mamifero': {
            'onivoro': 'homem',
            'herbivoro': 'vaca'
        }
    },
    'invertebrado': {
        'inseto': {
            'hematofago': 'pulga',
            'herbivoro': 'lagarta'
        },
        'anelideo': {
            'hematofago': 'sanguessuga',
            'onivoro': 'minhoca'
        }
    }
}

if vertebra in resultados:
    if tipo in resultados[vertebra]:
        if alimentacao in resultados[vertebra][tipo]:
            print(resultados[vertebra][tipo][alimentacao])